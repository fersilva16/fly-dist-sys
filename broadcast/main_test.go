package main

import (
	"fly-dist-sys/testutils"
	"io"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	maelstrom "github.com/jepsen-io/maelstrom/demo/go"
	"github.com/stretchr/testify/require"
)

func TestTopology1(t *testing.T) {
	require := require.New(t)

	var stdin io.WriteCloser
	var stdout io.ReadCloser

	node, stdin, stdout = testutils.NewNode()

	go main()

	err := testutils.InitNode(stdin, stdout, "n0", []string{"n0"})

	require.NoError(err)

	output, err := testutils.RPC(stdin, stdout, TopologyRequest{
		MessageBody: maelstrom.MessageBody{
			Type:  "topology",
			MsgID: 2,
		},

		Topology: map[string][]string{"n0": {}},
	})

	require.NoError(err)

	snaps.MatchSnapshot(t, output)
	snaps.MatchJSON(t, neighbours)
}

func TestTopology2(t *testing.T) {
	require := require.New(t)

	var stdin io.WriteCloser
	var stdout io.ReadCloser

	node, stdin, stdout = testutils.NewNode()

	go main()

	err := testutils.InitNode(stdin, stdout, "n0", []string{"n0", "n1", "n2", "n3", "n4"})

	require.NoError(err)

	output, err := testutils.RPC(stdin, stdout, TopologyRequest{
		MessageBody: maelstrom.MessageBody{
			Type:  "topology",
			MsgID: 2,
		},

		Topology: map[string][]string{
			"n0": {"n3", "n1"},
			"n1": {"n4", "n2", "n0"},
			"n2": {"n1"},
			"n3": {"n0", "n4"},
			"n4": {"n1", "n3"},
		},
	})

	require.NoError(err)

	snaps.MatchSnapshot(t, output)
	snaps.MatchJSON(t, neighbours)
}
