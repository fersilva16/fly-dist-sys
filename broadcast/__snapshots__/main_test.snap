
[TestSingle - 1]
{"src":"n0","dest":"c0","body":{"in_reply_to":2,"type":"topology_ok"}}
---

[TestSingle - 2]
{"src":"n0","dest":"c0","body":{"in_reply_to":2,"type":"broadcast_ok"}}
---

[TestSingle - 3]
{"src":"n0","dest":"c0","body":{"in_reply_to":2,"messages":[1],"type":"read_ok"}}
---

[TestMulti - 1]
{"src":"n0","dest":"c0","body":{"in_reply_to":2,"type":"topology_ok"}}
---

[TestMulti - 2]
{"src":"n0","dest":"c0","body":{"in_reply_to":3,"type":"broadcast_ok"}}
---

[TestMulti - 3]
{"src":"n0","dest":"n3","body":{"message":1,"msg_id":1,"type":"broadcast"}}
---

[TestMulti - 4]
{"src":"n0","dest":"n1","body":{"message":1,"msg_id":2,"type":"broadcast"}}
---

[TestMulti - 5]
{"src":"n0","dest":"c0","body":{"in_reply_to":4,"messages":[1],"type":"read_ok"}}
---
